src/server/dflycmd.cc

```diff
@@ -428,7 +428,7 @@ uint32_t DflyCmd::CreateSyncSession(ConnectionContext* cntx) {
   unique_lock lk(mu_);
   unsigned sync_id = next_sync_id_++;
 
-  unsigned flow_count = shard_set->size() + 1;
+  unsigned flow_count = shard_set->size() == 1 ? 1 : shard_set->size() + 1;
```

Comment:
Not sure what comment to add:

"Only on a single thread the io fiber does not occupy a separare thread"?

Reply:
I would implement it differently. Just use "pool->size()". You can access thread pool via shardset imho

Reply:
Indeed 😅